modules/api/pkg/resources/machine/common.go

```diff
@@ -273,30 +274,48 @@ func getVSphereProviderSpec(c *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc
 	return EncodeAsRawExtension(config)
 }
 
-func getBaremetalProviderSpec(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*runtime.RawExtension, error) {
-	// Check if Tinkerbell configuration is provided
+func GetBaremetalProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*baremetal.RawConfig, error) {
+	// Ensure Tinkerbell configuration is provided
 	if nodeSpec.Cloud.Baremetal.Tinkerbell == nil {
-		return nil, errors.New("tinkerbell provisioner is required")
+		return nil, errors.New("tinkerbell provisioner is required in baremetal configuration")
 	}
 
-	// Create TinkerbellNodeSpec from nodeSpec
-	tinkerbellSpec := apiv1.TinkerbellNodeSpec{
-		OsImageUrl:  nodeSpec.Cloud.Baremetal.Tinkerbell.OsImageUrl,
-		HardwareRef: nodeSpec.Cloud.Baremetal.Tinkerbell.HardwareRef,
+	tinkerbellConfig := nodeSpec.Cloud.Baremetal.Tinkerbell
+
+	// Validate required fields in Tinkerbell configuration
+	if tinkerbellConfig.HardwareRef.String() == "/" || tinkerbellConfig.OsImageUrl == "" {
+		return nil, errors.New("HardwareRef and OsImageUrl must be provided in Tinkerbell configuration")
+	}
+
+	// Prepare Tinkerbell plugin specification
+	tinkerbellSpec := &tink.TinkerbellPluginSpec{
+		ClusterName: providerconfig.ConfigVarString{Value: cluster.Name},
+		HardwareRef: tinkerbellConfig.HardwareRef,
+		Auth:        tink.Auth{Kubeconfig: providerconfig.ConfigVarString{Value: cluster.Spec.Cloud.Baremetal.Tinkerbell.Kubeconfig}},
+		OSImageURL:  providerconfig.ConfigVarString{Value: tinkerbellConfig.OsImageUrl},
 	}
 
-	// Marshal TinkerbellNodeSpec into JSON
-	tinkerbellData, err := tinkerbellSpec.MarshalJSON()
+	// Marshal TinkerbellSpec into JSON
+	tinkerbellData, err := json.Marshal(tinkerbellSpec)
 	if err != nil {
-		return nil, fmt.Errorf("failed to marshal TinkerbellNodeSpec: %w", err)
+		return nil, fmt.Errorf("failed to marshal TinkerbellPluginSpec: %w", err)
 	}
 
 	// Create a new baremetal.RawConfig and set the DriverSpec
 	config := &baremetal.RawConfig{
 		DriverSpec: runtime.RawExtension{Raw: tinkerbellData},
+		Driver:     providerconfig.ConfigVarString{Value: "tinkerbell"},
```

Comment:
Why is the value being hardcoded here? Can't we use a constant from machine-controller for this?